tests/generate_template_reference_data.py

```diff
@@ -170,79 +180,84 @@ def main(args):
         init(p)
-        fp = get_masked_flowproject(p)
-        # Here we set the appropriate executable for all the operations. This
-        # is necessary as otherwise the default executable between submitting
-        # and running could look different depending on the environment.
-        executable = "/usr/local/bin/python"
-        for group in fp.groups.values():
-            for op_key in group.operations:
-                if op_key in group.operation_directives:
-                    group.operation_directives[op_key]["executable"] = executable
-        for job in fp:
-            with job:
-                kwargs = job.statepoint()
-                env = get_nested_attr(flow, kwargs["environment"])
-                fp._environment = env
-                parameters = kwargs["parameters"]
-                if "bundle" in parameters:
-                    bundle = parameters.pop("bundle")
-                    fn = "script_{}.sh".format("_".join(bundle))
-                    tmp_out = io.TextIOWrapper(io.BytesIO(), sys.stdout.encoding)
-                    with redirect_stdout(tmp_out):
-                        try:
-                            fp.submit(
-                                jobs=[job],
-                                names=bundle,
-                                pretend=True,
-                                force=True,
-                                bundle_size=len(bundle),
-                                **parameters,
-                            )
-                        except jinja2.TemplateError as e:
-                            print("ERROR:", e)  # Shows template error in output script
-
-                    # Filter out non-header lines
-                    tmp_out.seek(0)
-                    with open(fn, "w") as f:
-                        with redirect_stdout(f):
-                            print(tmp_out.read(), end="")
-                else:
-                    for op in {**fp.operations, **fp.groups}:
-                        if "partition" in parameters:
-                            # Don't try to submit GPU operations to CPU partitions
-                            # and vice versa.  We should be able to relax this
-                            # requirement if we make our error checking more
-                            # consistent.
-                            if operator.xor(
-                                "gpu" in parameters["partition"].lower(),
-                                "gpu" in op.lower(),
-                            ):
-                                continue
-                        fn = f"script_{op}.sh"
+        with get_masked_flowproject(p) as fp:
+            # Here we set the appropriate executable for all the operations. This
+            # is necessary as otherwise the default executable between submitting
+            # and running could look different depending on the environment.
+            executable = "/usr/local/bin/python"
```

Comment:
Save this as a module-level variable `MOCK_EXECUTABLE` that you can use both here and in the `get_mocked_flowproject` context above.

Reply:
Very nice refactor. Good find.